packages/account/src/Components/additional-kyc-info-modal/form-config.tsx

```diff
@@ -0,0 +1,308 @@
+                    value: 'Speculative',
+                },
+            ],
+            rules: [
+                [
+                    'req',
+                    <Localize key='account_opening_reason' i18n_default_text='Intended use of account is required.' />,
+                ],
+            ],
+        },
+    };
+    return config;
+};
+
+/**
+ * Generate initial values for form
+ *
+ * @param fields - TFormFieldsConfig
+ * @returns Record<string, unknown> - initial values for form
+ */
+const generateInitialValues = (fields: ReturnType<typeof getFormFieldsConfig>) => {
+    const initial_values: Record<TFields, string> = {} as Record<TFields, string>;
+    (Object.keys(fields) as TFields[]).forEach(field => {
+        initial_values[field] = fields[field].initial_value;
+    });
+    return initial_values;
+};
+
+/**
+ * This function is used to transform form fields config to the format that is used in Formik or Formik Field
+ *
+ * @param fields - TFormFieldsConfig
+ * @param name - TFields
+ * @param with_input_types - boolean
+ * @returns - TGetField
+ */
+const getField = (fields: TFormFieldsConfig, name: TFields, with_input_types: boolean): TGetField => {
+    const { label, placeholder, required, disabled, type, list_items } = fields[name];
+
+    return {
+        name,
+        label,
+        required,
+        disabled,
+        ...(with_input_types ? { type } : {}),
+        ...(placeholder ? { placeholder } : {}),
+        ...(list_items ? { list_items } : {}),
+    };
+};
+
+/**
+ * Function to transform and return form config that can be used within the component that renders the form
+ *
+ * @param options.account_settings - GetSettings - account settings from API
+ * @param options.residence_list - ResidenceList - residence list from API
+ * @param options.required_fields - TFields[] - required fields
+ * @param options.with_input_types - boolean - to include input types in the form config or not
+ * @returns
+ */
+export const getFormConfig = (options: {
+    account_settings: GetSettings;
+    residence_list: ResidenceList;
+    required_fields: TFields[];
+    with_input_types?: boolean;
+}) => {
+    const { account_settings, residence_list, required_fields, with_input_types = false } = options;
+    const fields_config = getFormFieldsConfig(account_settings, residence_list, required_fields);
+    const inputs: Record<TFields, TGetField> = {} as Record<TFields, TGetField>;
+    Object.keys(fields_config).forEach(field_key => {
+        // @ts-expect-error `field_key` is always a key of `fields_config`, Hence can ignore the TS error.
+        inputs[field_key] = getField(fields_config, field_key, with_input_types);
+    });
+    return {
+        fields: inputs,
+        /** typing fields_config as any as this current config has different structure
+         * and generateValidationFunction should have generic types
+         * */
+        validate: generateValidationFunction('', fields_config as any),
```

Comment:
May be we can add a TODO to take a look at the typing of`generateValidationFunction` since its used in Accounts package only

Reply:
TODOs will be left there forever also the types are not generic for this function. It has been used in alot so it'll be hard to refactor this function as well